src/content/docs/release-notes/agent-release-notes/go-release-notes/go-agent-3-24-1.mdx

```diff
@@ -0,0 +1,35 @@
+---
+subject: Go agent
+releaseDate: '2023-08-10'
+version: 3.24.1
+downloadLink: 'https://github.com/newrelic/go-agent/tree/v3.24.1'
+features: []
+bugs: ["Performance improvement around calls to security agent. In some cases, unnecessary setup operations were being performed even if there was no security agent present to use that"]    
+security: ["Bumped nrmongo to v1.1.1","Bumped nrawssdk-v1 to 1.1.2","Bumped nrawssdk-v2 to 1.2.2","Bumped nrecho-v3 to 1.0.2","Bumped nrecho-v4 to 1.0.4",""Bumped nrhttprouter to 1.3.0","Bumped nrlambda to v1.2.2","Bumped nrnats to 1.1.5","Bumped nrreddis-v8 to v1.0.1"]
+---
+
+<Callout variant="important">
+We recommend updating to the latest agent version as soon as it's available. If your organization has established practices that prevent you from upgrading to the latest version, ensure that your agents are regularly updated to a version that's at most 90 days old. Read more about [keeping your agent up to date](/docs/new-relic-solutions/new-relic-one/install-configure/update-new-relic-agent/).
+</Callout>
+
+## 3.24.1
+### Fixed
+ * Performance improvement around calls to security agent. In some cases, unnecessary setup operations were being performed even if there was no security agent present to use that. These are now conditional on the security agent being present in the application (note that this will enable the setup code if the security agent is *present* in the application, regardless of whether it’s currently enabled to run). This affects:
+    * Base agent code (updated to v3.24.1)
+    * `nrmongo` integration (updated to v1.1.1)
+ * Fixed unit tests for integrations which were failing because code level metrics are enabled by default now:
+    * `nrawssdk-v1` (updated to v1.1.2)
+    * `nrawssdk-v2` (updated to v1.2.2)
+    * `nrecho-v3` (updated to v1.0.2)
+    * `nrecho-v4` (updated to v1.0.4)
+    * `nrhttprouter` (updated to v1.3.0)
+    * `nrlambda` (updated to v1.2.2)
+    * `nrnats` (updated to v1.1.5)
+    * `nrredis-v8` (updated to v1.0.1)
+### Changed
+ * Updated all integration `go.mod` files to reflect supported Go language versions.
+### Support statement
+We use the latest version of the Go language. At minimum, you should be using no version of Go older than what is supported by the Go team themselves (i.e., Go versions 1.19 and later are supported).
+We recommend updating to the latest agent version as soon as it’s available. If you can’t upgrade to the latest version, update your agents to a version no more than 90 days old. Read more about keeping agents up to date. (https://docs.newrelic.com/docs/new-relic-solutions/new-relic-one/install-configure/update-new-relic-agent/)
+See the [Go agent EOL Policy](/docs/apm/agents/go-agent/get-started/go-agent-eol-policy/) for details about supported versions of the Go agent and third-party components.
```

Comment:
```suggestion
<Callout variant="important">
We recommend updating to the latest agent version as soon as it's available. If your organization has established practices that prevent you from upgrading to the latest version, ensure that your agents are regularly updated to a version that's at most 90 days old. Read more about [keeping your agent up to date](/docs/new-relic-solutions/new-relic-one/install-configure/update-new-relic-agent/).
</Callout>

## 3.24.1
### Fixed
 * Performance improvement around calls to the security agent. In some cases, unnecessary setup operations were being performed even if there was no security agent present to use that. These are now conditional on the security agent being present in the application (note that this will enable the setup code if the security agent is *present* in the application, regardless of whether it’s currently enabled to run). This affects:
    * Base agent code (updated to v3.24.1)
    * `nrmongo` integration (updated to v1.1.1)
 * Fixed unit tests for integrations that were failing because code-level metrics are enabled by default now:
    * `nrawssdk-v1` (updated to v1.1.2)
    * `nrawssdk-v2` (updated to v1.2.2)
    * `nrecho-v3` (updated to v1.0.2)
    * `nrecho-v4` (updated to v1.0.4)
    * `nrhttprouter` (updated to v1.3.0)
    * `nrlambda` (updated to v1.2.2)
    * `nrnats` (updated to v1.1.5)
    * `nrredis-v8` (updated to v1.0.1)
### Changed
 * Updated all integration `go.mod` files to reflect supported Go language versions.
### Support statement
We use the latest version of the Go language. At a minimum, you should be using no version of Go older than what is supported by the Go team themselves (i.e., Go versions 1.19 and later are supported).
We recommend updating to the latest agent version as soon as it’s available. If you can’t upgrade to the latest version, update your agents to a version no more than 90 days old. Read more about keeping agents up to date. (https://docs.newrelic.com/docs/new-relic-solutions/new-relic-one/install-configure/update-new-relic-agent/)
See the [Go agent EOL Policy](/docs/apm/agents/go-agent/get-started/go-agent-eol-policy/) for details about supported versions of the Go agent and third-party components.
```